src/main/scala/org/constellation/util/Distance.scala

```diff
@@ -0,0 +1,18 @@
+package org.constellation.util
+
+import com.google.common.hash.Hashing
+import org.constellation.primitives.Schema.Id
+
+object Distance {
+  def calculate(hash: String, id: Id): BigInt =
+    numeric256(id.toPublicKey.getEncoded) ^ numeric256(hash.getBytes())
+
+  def calculate(id1: Id, id2: Id): BigInt =
+    numeric256(id1.toPublicKey.getEncoded) ^ numeric256(id2.toPublicKey.getEncoded)
+
+  private def numeric256(hash: Array[Byte]): BigInt = {
+    val sha256 = Hashing.sha256.hashBytes(hash).asBytes()
+    BigInt(sha256)
```

Comment:
Is the `xor` on on the `BigInt` the same as a byte-wise `xor` of the fixed length objects you get before the cast? In other words, is this line necessary?

I'd call it `numericSha256` btw.

Reply:
I believe it is, but it's probably preferable to do the XOR on the bytes, this was mostly just a proof of concept for sorting. 

As long as we hash entries before BigInt they should be roughly the same 'size'. (i.e. in the space of output sha256.) But it should be equivalent to doing XOR on Array[Byte] and then converting to BigInt before sorting.